rest/utilities_testing_blip_client.go

```diff
@@ -571,28 +589,67 @@ func (btr *BlipTesterReplicator) initHandlers(btc *BlipTesterClient) {
 			doc.lock.Lock()
 			defer doc.lock.Unlock()
 
-			var newVersion DocVersion
+			var incomingVersion DocVersion
+			var versionToWrite DocVersion
 			var hlv db.HybridLogicalVector
+			isDelete := true
 			if btc.UseHLV() {
+				var incomingHLV *db.HybridLogicalVector
 				if revHistory != "" {
-					existingVersion, _, err := db.ExtractHLVFromBlipMessage(revHistory)
-					require.NoError(btr.TB(), err, "error extracting HLV %q: %v", revHistory, err)
-					hlv = *existingVersion
+					incomingHLV, err = db.FromHistoryForHLV(revHistory)
+					require.NoError(btr.TB(), err, "error extracting HLV history %q: %v", revHistory, err)
+					hlv = *incomingHLV
 				}
-				v, err := db.ParseVersion(revID)
+				incomingCV, err := db.ParseVersion(revID)
 				require.NoError(btr.TB(), err, "error parsing version %q: %v", revID, err)
-				newVersion = DocVersion{CV: v}
-				require.NoError(btr.TB(), hlv.AddVersion(v))
+				incomingVersion = DocVersion{CV: incomingCV}
+
+				// fetch client's latest version to do conflict check and resolution
+				latestClientRev, err := doc._latestRev()
+				require.NoError(btc.TB(), err, "couldn't get latest revision for doc %q", docID)
+				if latestClientRev != nil {
+					clientCV := latestClientRev.version.CV
+
+					// safety check - ensure SG is not sending a rev that we already had - ensures changes feed messaging is working correctly to prevent
+					if clientCV.SourceID == incomingCV.SourceID && clientCV.Value == incomingCV.Value {
+						require.FailNow(btc.TB(), "incoming revision %v is equal to client revision %v - should've been filtered via changes response before ending up as a rev", incomingCV, clientCV)
```

Comment:
I'm about to throw up a PR to check this with ruleguard, but `FailNow` isn't fmt.Sprintf CBG-4455 to not panic- this needs to be:

```suggestion
						require.FailNow(btc.TB(), fmt.Sprintf("incoming revision %v is equal to client revision %v - should've been filtered via changes response before ending up as a rev", incomingCV, clientCV))
```

Reply:
fixed all instances of this